### PR TITLE
db_postgres doco update and SqlQuery parameter substitution (rebased)

### PR DESCRIPTION
**Pull request by jlp765**
(Rebase of #3334 )

Doco update to match the other db_xxx modules.

HOWEVER: db_postgres is slightly different, because there are two ways of doing SQL parameter substitution: the Postgres way using SqlPrepared (numbered parameters with the prepare() proc), and using the SqlQuery like the other modules using "?" as the identifier.

So an extra note about this added to the doco, and extra procs added where there were not already two of each. (I haven't worked out how to handle tryInsertId() and insertId() using SqlPrepared, so not all procs handle both substitution methods)

### CODE REVIEW

**Comment by dom96** [comment]
The point of the db_* modules is that they are all compatible. Postgres should support `?` and only `?`.

**Comment by jlp765** [comment]
So do I close #3342, and when the db_postgres module is rolled back to what it was, then submit a PR for the doco changes ?

**Comment by Araq** [comment]
I think this solution is just fine? postgres supports both ? and $1 then and if I want compatibility to other db* modules I must use ?. Did I miss anything?

**Comment by dom96** [comment]
Alright. Let's clean up this PR a bit then.

File: lib/impure/db_postgres.nim [hunk 1]
```diff
@@ -9,7 +9,61 @@
 
 ## A higher level `PostgreSQL`:idx: database wrapper. This interface
 ## is implemented for other databases too.
-
+##
+##  **Note**: There are two approaches to **parameter substitution** in this module
+##  1.  ``SqlQuery`` using ``?,?,?,...`` (same as the other db_xxxx modules)
+##
+##  .. code-block:: Nim
+##   ``sql"INSERT INTO myTable (colA,colB,colC) VALUES (?,?,?)"``
```
Comment (dom96) [inline lib/impure/db_postgres.nim]: Please remove the `` here, they are not needed as far as I can see.

**Comment by dom96** [comment]
I think the rest is fine,  are you happy @Araq or do you need to review?

**Comment by Araq** [comment]
Cannot review this. This needs tests!

**Comment by jlp765** [comment]
(I stuffed up trying to get this up-to-date with the latest devel head)

On the subject of testing, **how do you see these db_xxxx modules being tested in the tests directory using koch** ?

Do we specify that there needs to be both mysql & postgres databases setup to provide a `test` database with user `nim` & password `test` (or something similar).
For sqlite, there just needs to be the sqlite3 dll/so/lib added to the bin directory or in the path ??

And where is the testing framework documented? 
I haven't found it on the nim-lang website.
It would be good to spell out a bit clearer what options (TSpec) can be added to the header of test nim files, with a few examples.

**Comment by Araq** [comment]
* koch only tests that these modules compile, they have no runtime tests.
* since they have external dependencies the db* modules should really move out of the stdlib, into a nimble package.
* the tester is documented here: https://github.com/nim-lang/Nim/blob/devel/contributing.rst

**Comment by dom96** [comment]
Merged manually. Not sure why Github didn't detect it.

**Pull request closed without merging**